File: Backend/src/main/java/com/resumebuilder/backend/dto/ExperienceDTO.java
package com.resumebuilder.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.validation.constraints.NotBlank;
import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ExperienceDTO {

    private Long id;

    @NotBlank(message = "Company is required")
    private String company;

    @NotBlank(message = "Title is required")
    private String title;

    private LocalDate startDate;
    private LocalDate endDate;
    private Boolean isCurrentRole;
    private String location;
    private String type;
    private List<String> bulletPoints;
    private String description;
}

File: app/src/main/java/com/snehil/cvoptima/data/remote/model/ExperienceDto.kt
package com.snehil.cvoptima.data.remote.model

import com.google.gson.annotations.SerializedName

data class ExperienceDto(
    @SerializedName("id") val id: Long? = null,
    @SerializedName("company") val company: String,
    @SerializedName("title") val title: String,
    @SerializedName("startDate") val startDate: String?,
    @SerializedName("endDate") val endDate: String?,
    @SerializedName("isCurrentRole") val isCurrentRole: Boolean,
    @SerializedName("description") val description: String?,
    @SerializedName("location") val location: String? = null,
    @SerializedName("type") val type: String? = null,
    @SerializedName("bulletPoints") val bulletPoints: List<String>? = null
)
